Source/Libraries/FaultData/DataResources/FaultDataResource.cs

```diff
@@ -1023,14 +1023,80 @@ private void PopulateFaultInfo(Fault fault, DataGroup dataGroup, VICycleDataGrou
+                    for (int index = 0; index < line.Path.Count; index++)
+                    {
+                        // We can break early if this condition is found because any segments after are shorter
+                        if (summary.Distance > line.Path[index].Length) break;
+                        Fault.Path forwardPath = new Fault.Path();
+                        forwardPath.TransmissionPath = line.Path[index];
+                        forwardPath.PathNumber = index;
+                        forwardPath.FaultDistance = summary.Distance;
+                        forwardPath.IsMain = false;
+                        // Skip if path doesn't exist
+                        bool useForward = true;
+                        bool useBackward = false;
+                        if (forwardPath.TransmissionPath.Segments.Count == 0) continue;
+
+                        // If we have exactly one segment then a backwards path won't be neccessary
+                        if (forwardPath.TransmissionPath.Segments.Count != 1)
+                        {
+                            bool forwardlyChiral= meterTable.QueryRecordsWhere(testQuery, meter.ID, forwardPath.TransmissionPath.Segments[0].ID).Any();
+                            bool backwardlyChiral = meterTable.QueryRecordsWhere(testQuery, meter.ID, forwardPath.TransmissionPath.Segments[forwardPath.TransmissionPath.Segments.Count - 1].ID).Any();
+                            useForward = forwardlyChiral || !backwardlyChiral;
+                            useBackward = !forwardlyChiral || backwardlyChiral;
+                        }
 
+                        if (useForward)
+                        {
+                            forwardPath.IsMain = 0 == index;
+                            double distanceRemaining = forwardPath.FaultDistance;
+                            foreach (LineSegment segment in forwardPath.TransmissionPath.Segments)
+                            {
+                                if (distanceRemaining < segment.Length)
+                                {
+                                    forwardPath.SegmentDistance = distanceRemaining;
+                                    forwardPath.FaultSegment = segment;
+                                    break;
+                                }
+                                distanceRemaining -= segment.Length;
+                            }
+                            faultPaths.Add(forwardPath);
+                        }
+                        if (useBackward)
+                        {
+                            Fault.Path backwardPath = forwardPath.ShallowCopy();
+                            backwardPath.IsMain = (0 == index) && !useForward;
+                            double distanceRemaining = backwardPath.FaultDistance;
+                            for (int pathIndex = backwardPath.TransmissionPath.Segments.Count - 1; pathIndex >= 0; pathIndex--)
```

Comment:
Consider using `foreach (LineSegment segment in backwardPath.TransmissionPath.Segments.Reverse())`.

Reply:
I didn't know this was a thing, thank you for the lesson!